### PR TITLE
8 update article

### PR DESCRIPTION
**Pull request by simonhighfield**

### CODE REVIEW

**Comment by simonhighfield** [comment]
I have a question about my error handling for 404 errors. This probably ggoes back to previous questions.

But in my models. js there are two functions, checkIfArticleExists, and fethArticleById that return a promise reject with the value {msg: endpoint not found}. I don't actually use the value of that promise in my error handling. Instead, a 404 gets passed only if none of the error handlng middleware gets triggered e.g. if the error code does not matche 22P02, next() is invoked, and that hops to the app.all() whih sends the 404

My question is: should I use I be using the error msg I send back from models.js and if so how? OR 
Or do I even need to sed a value in that promise reject - would it be possible to just send promse.reject() - leaving the bracket empty?

Thanks, Simon

**Comment by roisinoreilly** [comment]
to try and answer your question, I don't think any of your error handling middleware is set up to use the msg key at any point. If you follow the promise chain, it's rejected in the model with the msg for example here:
`return Promise.reject({msg: 'endpoint not found'})`
this passes error to the catch, which passes it to the middleware. It now has the status key of 200 and the key of msg somewhere with your custom message. Have a look at this section of the notes on custom errors https://notes.northcoders.com/courses/js-back-end/error-handling#custom-errors but essentially I think you're not using the msg anywhere. I'd also recommend doing it how it does in the notes and sending back a key of status in the Promise.reject as it does there. 

I wouldn't recommend just invoking Promise.reject() by itself, it gives you no way of differentiating it from other errors and it isn't useful in sending back specific status codes and messages. Any questions about any of this though let me know!

File: MVC/controller.js [hunk 1]
```diff
@@ -62,4 +63,24 @@ exports. postComment = (req, res, next) => {
...
+
+    Promise.all([fetchVotes(article_id), checkIfArticleExists(article_id)])
+    // the order of the functions above affects the array deconstruction below
+    .then(([result]) => {
+        const { votes } = result
+        const newVotes = votes + inc_votes
+        return setVotes(article_id, newVotes)   
+    })
+    .then((updatedArticle) => {
+        res.status(200).send(updatedArticle)
```
Comment (roisinoreilly) [inline MVC/controller.js:82]: don't forget to send back on an object

**Review by roisinoreilly — COMMENTED** [review]
I'd suggest re-organising your file structure at this point. My recommendation would be that app should always be in the root for most projects, then I would create a folder each called models and controllers (doesn't have to be in an mvc folder, can be at root), and then starting to extract your models and controllers by endpoint so the files aren't so big ie. topics.controllers.js, articles.models.js etc.

other than that your code for this ticket is great and your tests are excellent! I would maybe suggest one more small test for if the object to be patched doesn't have the right keys or they're in the wrong format. Other than that though really nice work!

**Pull request merged**